legistar/bills.py

```diff
@@ -349,14 +349,31 @@ def relations(self, matter_id):
         else:
             return []
 
-    def text(self, matter_id):
+    def text(self, matter_id, latest_version_value=None):
+        '''Historically, we have determined the latest version of a bill
+        by finding the version with the highest value (either numerical or otherwise).
+
+        However, the `MatterVersion` field on the matter detail page
+        most accurately identifies the latest version of a bill.
+        This proves to be true for Metro, in particular.
+
+        Other municipalities may share this characteristic with Metro.
+        Until we know more, the `text` function accepts `latest_version_value`,
+        i.e., matter['MatterVersion'], as an optional argument.
+        '''
         version_route = '/matters/{0}/versions'
         text_route = '/matters/{0}/texts/{1}'
 
         versions = self.endpoint(version_route, matter_id)
 
-        latest_version = max(
-            versions, key=lambda x: self._version_rank(x['Value']))
+        if latest_version_value:
```

Comment:
how do we know what the latest_version_value is?

Reply:
We find it in the matter dict returned from scraping the API. Like this: https://github.com/opencivicdata/scrapers-us-municipal/blob/optional-arg-text/lametro/bills.py#L295

Reply:
It would be better to do that all in this method.

Reply:
Sure, we could do that by calling [`matter`](https://github.com/opencivicdata/python-legistar-scraper/blob/master/legistar/bills.py#L254), but that adds to the number of times we're hitting legistar, and it duplicates the efforts of the scraper, e.g., the scrape already calls `matters` https://github.com/opencivicdata/scrapers-us-municipal/blob/master/lametro/bills.py#L148-L156

Unless...do you have another solution in mind for "doing that all in this method"?

Reply:
you are right that it adds another call, but it makes the method a lot more intuitive. I think it's probably worth it.

I do think that there's probably a better architecture for this this API along the lines of

```python
class Matter(object):

    __init__(self, matter_id):
        self._d = request.get(self.matter_endpoint(matter_id))

   def text(self):
       matter_version = self._d['MatterVersion']
       ...
       return most_recent_tex
```
  
This would avoid the duplicative call and reduce the extend of matter_id passing we are doing here.

In any case, that's well beyond the scope of this PR.

Reply:
Oh, poo. I just pushed a revision...actually, could you give your opinion on my latest commit? I think the code is more intuitive, but it assumes that other municipalities use the `MatterVersion` field exactly as Metro does. (I'd love to assume that they do, but can we be sure?)

Reply:
i would check to see if _version_rank is definited as an attribute. If it is, then use that.

Reply:
okay